### PR TITLE
Feature/edit aatf contact

### PR DESCRIPTION
**Pull request by JB12298**

### CODE REVIEW

File: src/EA.Weee.Database/scripts/Update/Release 2/Sprint8/20190508-0900-UpdateAATFTable.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+           ,[Postcode]
+           ,[CountryId]
+           ,[Telephone]
+           ,[Email])
+     VALUES
+           (@defaultContactId
+           ,'First Name'
+           ,'Last Name'
+           ,'Director'
+           ,'1 Address Lane'
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Database/scripts/Update/Release 2/Sprint8/20190508-0900-UpdateAATFTable.sql]: Remove the transaction so we don't have same issue as Will's :)

File: src/EA.Weee.Domain.Tests.Unit/AatfReturn/AatfContactTests.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        {
+            Action constructor = () =>
+            {
+                var @return = new AatfContact(firstName, lastName, position, address, A.Dummy<string>(), town, A.Dummy<string>(), A.Dummy<string>(), A.Dummy<Country>(), telephone, email);
+            };
+
+            constructor.Should().Throw<ArgumentException>();
+        }
+    }
+}
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Domain.Tests.Unit/AatfReturn/AatfContactTests.cs]: Add test to check that the properties passed are mapped.

Possible to aldo add test for the update contact method.

File: src/EA.Weee.RequestHandlers/AatfReturn/Internal/GetAatfContactHandler.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        {
+            authorization.EnsureCanAccessInternalArea();
+
+            var contact = await dataAccess.GetContact(message.AatfId);
+
+            var result = mapper.Map(contact);
+
+            result.CanEditContactDetails = authorization.CheckUserInRole(Roles.InternalAdmin);
+
+            return result;
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/AatfReturn/Internal/GetAatfContactHandler.cs]: Add Test to check that the result is the result of the mapped object
Add Test to check that the CanEditContactDetails property is set based on the authorization.CheckUserInRole

File: src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs [hunk 4]
```diff
@@ -76,6 +83,56 @@ public async Task<ActionResult> ManageAatfs(ManageAatfsViewModel viewModel)
...
+
+                    await client.SendAsync(User.GetAccessToken(), request);
+
+                    return RedirectToAction("Details", new { Id = viewModel.AatfId });
+                }
+            }
+
+            using (var client = apiClient())
+            {
+                viewModel.ContactData.AddressData.Countries = await client.SendAsync(User.GetAccessToken(), new GetCountries(false));
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs]: Not sure we have done this elsewhere, but if we have to re get the countries list can we create a unit test to make sure the model property is set based from the api return

File: src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/AatfDataToAatfDetailsViewModel.cs [hunk 5]
```diff
@@ -3,29 +3,40 @@
     using EA.Prsd.Core;
     using EA.Prsd.Core.Mapper;
     using EA.Weee.Core.AatfReturn;
+    using EA.Weee.Web.Areas.AatfReturn.Mappings.ToViewModel;
     using EA.Weee.Web.Areas.Admin.ViewModels.Aatf;
     using System;
 
-    public class AatfDataToAatfDetailsViewModel : IMap<AatfData, AatfDetailsViewModel>
+    public class AatfDataToAatfDetailsViewModel : IMap<AatfDataToAatfDetailsViewModelTransfer, AatfDetailsViewModel>
     {
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/AatfDataToAatfDetailsViewModel.cs]: Rename to AatfDataToAatfDetailsViewModelMapper

File: src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs (hunk 4 above)
Comment (JB12298) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs]: I don't check this specifically, but I check that countries request is made?  AatfControllerTests line 211

Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs]: Cool, if you can check that the API call is made and the view model contains the value returns from the api...

Comment (JB12298) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs]: Added a test

**Pull request merged**